### ISSUE TITLE
having issues with the mnet3 branch

### ISSUE THREAD

**Issue by justinjohn0306** [FORTH-ModelBasedTracker/MocapNET#115]
Here's a screenshot of the error i'm facing after the initialization of the mnet3 branch of the repo. The cmd line I used for the test: ``./MocapNET2CSV --from con0014/2dJoints_v1.4.csv --ik 0.001 15 30 --mt --show 3  --hands --label s10-p01-c0014-f --seriallength 4 --gdpr``

![Screenshot from 2023-12-08 11-31-34](https://github.com/FORTH-ModelBasedTracker/MocapNET/assets/34035011/41ca597c-3f29-471d-9d00-d0b9692fed19)

**Comment by AmmarkoV** [comment]
Unfortunately the error looks like something has internally changed in the way tensorflow restores and executes saved model signatures (?).

My MNET3 code relies on tensorflow CPP bindings :
https://www.tensorflow.org/api_docs/cc
using this wrapper :
https://github.com/FORTH-ModelBasedTracker/MocapNET/blob/master/src/Tensorflow2/tensorflow2.h

The MNET3 training data was produced in 2021 with TF 2.3.1 so this might mean that tensorflow has broken compatibility  https://www.tensorflow.org/guide/versions ?

If that is the case the fix would be to : 
1) dump the old models to a different format e.g. ONNX and add a wrapper for ONNX execution
2) Retrain the models from scratch and save them using the new convention, however these would be new models
3) somehow Upgrade TF2.3 models to the new version ?

The best I can offer at the time is the python MNET4 backend that has body + hands + face.. :( 
https://colab.research.google.com/github/FORTH-ModelBasedTracker/MocapNET/blob/mnet4/mocapnet4.ipynb

Unfortunately if this is a compatibility problem it is hard to fix

**Comment by AmmarkoV** [comment]
I think there is a problem with the Google Drive model files..!
I reverted back to the ammar.gr version of the files and I think it fixes the issue, please confirm!

but also keep in mind the new version : 
https://colab.research.google.com/github/FORTH-ModelBasedTracker/MocapNET/blob/mnet4/mocapnet4.ipynb

**Comment by AmmarkoV** [comment]
@justinjohn0306  I have also updated the Google Drive files and they work once again.
Will keep the issue open until you confirm!

**Comment by justinjohn0306** [comment]
> @justinjohn0306 I have also updated the Google Drive files and they work once again. Will keep the issue open until you confirm!

It works now but the output joints from the bvh file is messed up, as you can see in the image below

![Screenshot from 2023-12-10 12-06-47](https://github.com/FORTH-ModelBasedTracker/MocapNET/assets/34035011/063823c6-678b-4fb1-ab1a-d798a5695205)

**Comment by justinjohn0306** [comment]
> I think there is a problem with the Google Drive model files..! I reverted back to the ammar.gr version of the files and I think it fixes the issue, please confirm!
> 
> but also keep in mind the new version : https://colab.research.google.com/github/FORTH-ModelBasedTracker/MocapNET/blob/mnet4/mocapnet4.ipynb

I did test the mnet4 branch and it wasn't accurate with the pose estimation with the mediapipe module

**Comment by ArEnSc** [comment]
we are running into the same issues as @justinjohn0306 here

**Comment by ArEnSc** [comment]
@AmmarkoV

**Comment by AmmarkoV** [comment]
Since the issue with model package is resolved, and the inverted arms is another issue 
I am opening a new issue ticket ( https://github.com/FORTH-ModelBasedTracker/MocapNET/issues/116 ) and closing this as resolved
